extensions/amp-story/1.0/amp-story-page.js

```diff
@@ -528,6 +531,19 @@ export class AmpStoryPage extends AMP.BaseElement {
     ]);
   }
 
+  /** @override */
+  onLayoutMeasure() {
+    const {height, width} = this.getLayoutSize();
+    if (
+      !AmpStoryPage.prerenderAllowed(this.element) ||
```

Comment:
You're tying how many times we run the PAGE_SIZE to how many pages are prerendered, which is unrelated.
All we need is making sure this only runs once, which doesn't have anything to do with which page is initially visible or how many are prerendered.

```
AmpStoryPage {
  constructor() {
    this.isFirstPage_ = matches(this.element, 'amp-story-page:first-of-type');
  }
}
```

And you're done without storing DOM elements, tying business logic to internal behavior, or using static properties.